vent/controller/control_module.py

```diff
@@ -147,21 +147,32 @@ def get_sensors(self) -> SensorValues:
 
     def get_alarms(self) -> List[Alarm]:
         # Returns all alarms as a list
+        lock.acquire()
         ls = self.logged_alarms
         for alarm_key in self.active_alarms.keys():
             ls.append(self.active_alarms[alarm_key])
+        lock.release()
         return ls
 
     def get_active_alarms(self):
         # Returns only the active alarms
-        return self.active_alarms
+        lock.acquire()
+        active_alarms = self.active_alarms
```

Comment:
I think this is returning a reference to the alarms rather than a copy. We want to make sure to return copies of the list. Same for other places an alarm list is returned.